### PR TITLE
[TieredStorage] Remove the general-purposed TieredStorageWriter

### PR DESCRIPTION
**Pull request by yhchiang-sol**
#### Problem
tiered_storage/writer.rs was added when we planned to support multiple
tiers in the tiered-storage (i.e., at least hot and cold).  However, as we
changed our plan to handle cold accounts as state-compressed accounts,
we don't need a general purposed tiered-storage writer at this moment.

#### Summary of Changes
Remove tiered_storage/writer.rs as we currently don't have plans to develop cold storage.

#### Test Plan
Existing tiered-storage tests.

### CODE REVIEW

**Review by brooksprumo — COMMENTED** [review]
> tiered_storage/writer.rs was added when we planned to support multiple
tiers in the tiered-storage (i.e., at least hot and cold). However, as we
changed our plan to handle cold accounts as state-compressed accounts,
we don't need a general purposed tiered-storage writer at this moment.

Why does this not also apply to TieredStorageReader?

**Comment by yhchiang-sol** [comment]
> Why does this not also apply to TieredStorageReader?

We probably want to keep the TieredStorageReader so that we can provide some backward compatibility
in case we introduce a new way to store hot storage in the future.  But we only need to keep one write at a time so we can deprecate writer.rs.

**Comment by yhchiang-sol** [comment]
But at least I find TieredReadableAccount can be repurposed to HotAccount https://github.com/anza-xyz/agave/pull/218

File: accounts-db/src/tiered_storage/writer.rs [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-        file_path: impl AsRef<Path>,
-        format: &'format TieredStorageFormat,
-    ) -> TieredStorageResult<Self> {
-        Ok(Self {
-            storage: TieredStorageFile::new_writable(file_path)?,
-            format,
-        })
-    }
-
-    pub fn write_accounts<
```
Comment (brooksprumo) [inline accounts-db/src/tiered_storage/writer.rs:34]: Who—if anyone—called this function before? I'm guessing no one? Maybe only tests are writing accounts? And those are calling `HotStorageWriter::write_accounts()` directly?

Comment (yhchiang-sol) [reply accounts-db/src/tiered_storage/writer.rs:34]: No one in master.  (Previously both hot and cold storage called this function in my draft development)

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**